pyrevolve/revolve_bot/revolve_module.py

```diff
@@ -265,6 +323,38 @@ def to_sdf(self, tree_depth='', parent_link=None, child_link=None):
         parent_link.append(imu_sensor)
         return visual, collision, imu_sensor
 
+    def count_module_connections(self) -> int:
+        """
+        Count how many connections the module has.
+        Connected TouchSensor and BrickSensor Modules are ignored
+        :return: number of connections
+        """
+        # The CoreModule does not have a parent module
+        return super(CoreModule, self).count_module_connections() - 1
+
+    def is_folding(self) -> bool:
+        # parent slot is not supposed to be set
+        parent_slot = None
+        target_slot = None
+        for slot, child in self.iter_children():
+            # is the child slot occupied?
+            if child is None:
+                continue
+            # ignore TouchSensor and BrickSensor modules
+            if isinstance(child, TouchSensorModule) or \
+                    isinstance(child, BrickSensorModule):
+                continue
+            if parent_slot is None:
+                parent_slot = slot
+            elif target_slot is None:
+                target_slot = slot
+            else:
+                # More than 2 children, this is not a fold
+                return False
+        if parent_slot is None or target_slot is None:
```

Comment:
This will always trigger when only the parent_slot is filled in the first iteration that a viable child is set as the parent_slot at line 348.

Reply:
What do you mean? This if will trigger if I've found both a first and a second child, at least that was my intention.

Reply:
Let's say we are in the first iteration of this for loop and it found a viable child and we are saving the slot of this child at line 348. Then now we come to 354 and we say that this statement is true, since the target slot is still unfilled, then we return false, I do not think this is expected behavior.